stp/constraints.lisp

```diff
@@ -105,36 +105,35 @@
 ;;    2.5 vanishing constraints    ;;
 ;;                                 ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-(defconstraint callcode-impose-exists ()
-  (if-not-zero IS_CALLCODE
-               (eq! EXISTS 1)))
+
+(defconstraint    CREATE-type---debug---vanishing-constraints   (:guard    (is_create))
+                  (begin
+                    (vanishes!    GAS_HI)
+                    (vanishes!    GAS_LO)))
+
+(defconstraint    CALL-type---debug---non-value-transferring-opcodes-have-zero-value    ()
+                  (if-not-zero    (+    IS_DELEGATECALL    IS_STATICCALL)
+                                  (begin
+                                    (vanishes!    VAL_HI)
+                                    (vanishes!    VAL_LO))))
+
+(defconstraint    CALL-type---debug---account-existence-only-matters-for-CALL    ()
+                  (if-not-zero    (+    IS_CALLCODE    IS_DELEGATECALL    IS_STATICCALL)
+                                  (vanishes!    EXISTS)))
```

Comment:
Non `debug` "debug" constraints.

Reply:
? What do you mean. Why they have debug is their names if it's not a debug constraint ?

Reply:
I've done that in a few places ... having constraints with `---debug---` in their name. Typically for stuff that maybe shouldn't be debug after all.